General/OOD/SemanticConstructorOverloading/SemanticConstructorOverloading.Problem/Optional Constructor Parameter Collision/Good/SampleClass.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace SemanticConstructorOverloading.Problem.Optional_Constructor_Parameter_Collision.Good
+{
+    /// <summary>
+    /// Optional Constructor parameters cause problems when they are of the same data type.
+    /// This is a better implementation than the alternative, but it's still ugly, and 
+    /// not really recommended.
+    /// </summary>
+    public class SampleClass
+    {
+        // Public property assignments managed by constructor. In this use case, these will most
+        // likely be used for some kind of object configuration option.
+        public string StringValueA { get; set; }
+        public string StringValueB { get; set; }
+
+        /// <summary>
+        /// When two parameters of the same type are OPTIONAL in a constructor, specifying them as 
+        /// optional parameters is easier to code and generally better than specifying all
+        /// possible combinations of non-optional parameters. This is still an ugly solution, but
+        /// it's better than multiple overloaded constructors with parsing required to differentiate 
```

Comment:
> with parsing required to differentiate

Why is parsing mentioned? doesn't the same problem apply to int or bool, etc?

Reply:
I still have a hard time understanding what's the problem. Why do we need to parse, why do we need the checks (I understand, but I pretend to be a beginner)